### PR TITLE
Implement methods rename, moveTo, and copyTo

### PR DESCRIPTION
**Pull request by dongsupark**
Hi,

while implementing seafile-fuse-client, (https://github.com/dongsupark/seafile-fuse-client.git) I also added new methods in class _SeafDirentBase: rename(), moveTo(), and copyTo(). Please consider merging them into master.

Thanks,
Dongsu

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -18,8 +18,8 @@ def client():
 
 @pytest.yield_fixture(scope='function')
 def repo(client):
-    repo_name = 'tmp-测试资料库-%s' % randstring()
-    repo_desc = 'tmp, 一个测试资料库-%s' % randstring()
```
Comment (lins05) [inline tests/fixtures.py]: Why change it here? The reason of using Chinese characters in tests is to make sure the library can handle non-ascii characters.

File: functests.sh [hunk 2]
```diff
@@ -2,11 +2,16 @@
...
+[ -z "$SEAFILE_TEST_SERVER_ADDRESS" ] && \
+	export SEAFILE_TEST_SERVER_ADDRESS="http://127.0.0.1:8000"
+[ -z "$SEAFILE_TEST_USERNAME" ] && \
+	SEAFILE_TEST_USERNAME="test@seafiletest.com"
+[ -z "$SEAFILE_TEST_PASSWORD" ] && \
+	SEAFILE_TEST_PASSWORD="testtest"
+[ -z "$SEAFILE_TEST_ADMIN_USERNAME" ] && \
+	SEAFILE_TEST_ADMIN_USERNAME="admin@seafiletest.com"
+[ -z "$SEAFILE_TEST_ADMIN_PASSWORD" ] && \
+	SEAFILE_TEST_ADMIN_PASSWORD="adminadmin"
```
Comment (lins05) [inline functests.sh]: Any particular reason of the changes here?

File: seafileapi/files.py [hunk 3]
```diff
@@ -39,14 +39,34 @@ def delete(self):
...
     def moveTo(self, dst_dir, dst_repo=None):
-        pass
+        """Move filename to newname (also to a different directory)
+        """
+        url = '/api2/repos/%s/file/' % self.repo.id + querystr(p=self.path)
+        if dst_repo is None:
+            dst_repo = self.repo.id
+        postdata = {'operation': 'move', 'dst_repo': dst_repo, 'dst_dir': dst_dir}
+        resp = self.client.post(url, data=postdata)
+        self.id = resp.headers['oid']
```
Comment (lins05) [inline seafileapi/files.py]: You should also add new tests for the rename/copyTo/moveTo methods.

**Comment by lins05** [comment]
@dongsupark thanks for the PR! I've left some line comments.

File: tests/fixtures.py (hunk 1 above)
Comment (dongsupark) [inline tests/fixtures.py]: I see. I thought that they would be mistakes, as I couldn't read them. :-)
I removed the patch and pushed it into my git repo.

File: functests.sh (hunk 2 above)
Comment (dongsupark) [inline functests.sh]: Oh, that was my mistake. For some reason I thought the script didn't work. But it's actually working as expected, as I've just tested it again.
I removed the patch, and pushed it into my git repo.

**Comment by freeplant** [comment]
It is rebased and merged in https://github.com/haiwen/python-seafile/pull/9

**Pull request closed without merging**